extensions/pkg/controller/backupbucket/controller.go

```diff
@@ -62,7 +65,7 @@ func DefaultPredicates(ignoreOperationAnnotation bool) []predicate.Predicate {
 // Add creates a new BackupBucket Controller and adds it to the Manager.
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager, args AddArgs) error {
-	args.ControllerOptions.Reconciler = NewReconciler(args.Actuator)
+	args.ControllerOptions.Reconciler = NewReconciler(args.Actuator, args.ErrorCodeDetector)
```

Comment:
TBH, I don't get the motivation behind introducing this `ErrorCodeDetector` interface in each and every controller in the extensions library. This new touches a lot of places that hard to follow already as of now.

If your goal is to have a central error code detection mechanism per extension, I would rather go for wrapping `reconcile.Reconciler` and just ran the returned error (if any) through the detection.
In the extensions library, there should be a check whether the returned error contains an error code (for example, by using `errors.As` with `gardencorev1beta1helper.Coder` interface). If such code was attached to the error, then the controller in the library should set the status accordingly.